src/pages/news/news-detail/news-detail.js

```diff
@@ -1,70 +1,53 @@
-import React, { useEffect } from 'react';
-import { useSelector, useDispatch } from 'react-redux';
+import React, { useState } from 'react';
+import { useQuery } from '@apollo/client';
 import parse from 'html-react-parser';
 import PropTypes from 'prop-types';
 import { Card, Typography, CardContent, CardHeader, CardMedia } from '@material-ui/core';
-
-import { getArticle } from '../../../redux/news/news.actions';
+import { useTranslation } from 'react-i18next';
 import { useStyles } from './news-detail.style';
-import { IMG_URL, TIME_OPTIONS, HYPHEN } from '../../../configs';
-import { Loader } from '../../../components/loader/loader';
+import { IMG_URL, TIME_OPTIONS } from '../../../configs';
+import { getNewsById } from '../operations/news-queries';
+import errorOrLoadingHandler from '../../../utils/errorOrLoadingHandler';
 
 const NewsDetail = ({ match }) => {
-  const { article, loading, language } = useSelector(({ News, Language }) => ({
-    article: News.activeArticle,
-    loading: News.loading,
-    language: Language.language
-  }));
-
-  const dispatch = useDispatch();
+  const [article, setArticle] = useState({});
 
-  useEffect(() => {
-    const articleId = match.params.id;
-    dispatch(getArticle(articleId.split(HYPHEN)[0]));
-    window.scrollTo(0, 0);
-  }, [match.params.id, dispatch]);
+  const articleId = match.params.id;
+  const id = articleId.split('-')[0];
 
+  const { loading, error } = useQuery(getNewsById, {
+    onCompleted: (data) => setArticle(data.getNewsById),
+    variables: { id }
+  });
+  const { t, i18n } = useTranslation();
+  const language = i18n.language === 'ua' ? 0 : 1;
   const styles = useStyles();
 
-  if (loading || !article) {
-    return (
-      <div className={styles.center}>
-        <Loader />
-      </div>
-    );
-  }
-  if (article.text[language].value === null) {
-    return <h2>Sory, this article is not translated. Try to change language</h2>;
-  }
-  const newsTitle =
-    article.title.length !== 0 ? article.title[language].value : 'No title provided';
+  if (loading || error || !article._id) return errorOrLoadingHandler(error, loading);
```

Comment:
What will receive errorOrLoadingHandler if loading and error are false. How it will work ?